src/MongoDB.Driver.Core/Core/Connections/CommandEventHelper.cs

```diff
@@ -45,6 +46,12 @@ internal class CommandEventHelper
             "copydb"
         };
 
+        private static readonly ReadOnlyDictionary<string, Func<BsonDocument, bool>> __conditionallySecuritySensitiveCommands =
+                new ReadOnlyDictionary<string, Func<BsonDocument, bool>>(new Dictionary<string, Func<BsonDocument, bool>>(StringComparer.OrdinalIgnoreCase)
+                {
+                    {"isMaster", c => c.Contains("speculativeAuthenticate")}
```

Comment:
This comment is out of date now since Dima has LGTMed and we've gone with Robert's low tech  approach of not reflecting against the data structures. Resolving.